### DISCUSSION TITLE
Is Run call multi-thread safe?

### DISCUSSION THREAD

**Discussion by praveen-ramesh** [adoconnection/RazorEngineCore#117] [Q&A] [answered]
If we use the caching technique recommended [here ](https://github.com/adoconnection/RazorEngineCore), the **compiledTemplate.Run** will encounter concurrent execution as our code is deployed in a multi-threaded environment.

So, is the **Run** method multi-thread safe? Has it been heavily tested against in such an environment?

**Comment by adoconnection** [comment, marked as answer]
Hi, yes, Run is thread safe. It was not heavily tested by me, but it is supposed to be thread safe by design.
Run will create new ```  IRazorEngineTemplate instance``` for each call, thus it is safe to call Run by multiple  threads

https://github.com/adoconnection/RazorEngineCore/blob/ce41fffe192b72c24be589c2c1ba007c1b2f2c32/RazorEngineCore/RazorEngineCompiledTemplate.cs#L92
